packages/nx/src/plugins/js/lock-file/pnpm-parser.ts

```diff
@@ -158,6 +221,33 @@ function getNodes(
   const keyMap = new Map<string, Set<ProjectGraphExternalNode>>();
   const nodes: Map<string, Map<string, ProjectGraphExternalNode>> = new Map();
 
+  // Extract and pre-parse patch information from patchedDependencies section
+  const patchEntries: Array<{
+    packageName: string;
+    versionSpecifier: string | null; // null for name-only patches
+    hash: string;
+  }> = [];
+  if (data.patchedDependencies) {
+    for (const [specifier, patchInfo] of Object.entries(
+      data.patchedDependencies
+    )) {
+      if (patchInfo && typeof patchInfo === 'object' && 'hash' in patchInfo) {
+        // Extract package name (works for all cases now)
+        // patchedDependencies always use @ syntax (never v5 / syntax)
+        const packageName = extractNameFromKey(specifier, false);
+
+        // If specifier has a version, extract it using the helper
+        const versionSpecifier = getVersion(specifier, packageName) || null;
+
+        patchEntries.push({
+          packageName,
+          versionSpecifier,
+          hash: patchInfo.hash,
+        });
+      }
+    }
+  }
```

Comment:
Let's change this to a `Map` for easier lookups and slightly better perf:
```suggestion
  const patchEntriesByPackage = new Map<
    string,
    Array<{
      versionSpecifier: string | null; // null for name-only patches
      hash: string;
    }>
  >();
  if (data.patchedDependencies) {
    for (const [specifier, patchInfo] of Object.entries(
      data.patchedDependencies
    )) {
      if (patchInfo && typeof patchInfo === 'object' && 'hash' in patchInfo) {
        const packageName = extractNameFromKey(specifier, false);
        const versionSpecifier = getVersion(specifier, packageName) || null;

        if (!patchEntriesByPackage.has(packageName)) {
          patchEntriesByPackage.set(packageName, []);
        }
        patchEntriesByPackage.get(packageName).push({
          versionSpecifier,
          hash: patchInfo.hash,
        });
      }
    }
  }
```